creator-node/src/services/ContentNodeInfoManager.ts

```diff
@@ -0,0 +1,161 @@
+import type Logger from 'bunyan'
+
+import _ from 'lodash'
+
+import initAudiusLibs from './initAudiusLibs'
+import { logger as defaultLogger } from '../logging'
+import defaultRedisClient from '../redis'
+import { Redis } from 'ioredis'
+
+export type EthContracts = {
+  getServiceProviderList: (spType: string) => Promise<LibsServiceProvider[]>
+}
+export type LibsServiceProvider = {
+  owner: any // Libs typed this as any, but the contract has it as address
+  delegateOwnerWallet: any // Libs typed this as any, but the contract has it as address
+  endpoint: any // Libs typed this as any, but the contract has it as string
+  spID: number
+  type: string
+  blockNumber: number
+}
+export type ContentNodeFromChain = {
+  endpoint: string
+  delegateOwnerWallet: string
+}
+export interface IContentNodeInfoManager {
+  getMapOfSpIdToChainInfo: () => Promise<Map<number, ContentNodeFromChain>>
+  getMapOfCNodeEndpointToSpId: () => Promise<Map<string, number>>
+  getSerializedMapOfCNodeEndpointToSpId: () => Promise<string>
+  getSpIdFromEndpoint: (endpoint: string) => Promise<number | undefined>
+  getContentNodeInfoFromSpId: (
+    spId: number
+  ) => Promise<ContentNodeFromChain | undefined>
+}
+
+const SP_ID_TO_CHAIN_INFO_MAP_KEY = 'contentNodeInfoManagerSpIdMap'
+
+function ContentNodeInfoManager(
```

Comment:
i find this file super hard to read but maybe i'm just not familiar with TS
i remember seeing something about how  classes are also not recommended?

Reply:
I'll think over improving readability. classes aren't ideal which is actually specifically why I did this as a pure function. it's kinda like a class but it doesn't use the `class` keyword or share any state -- it just makes it so we don't have to call each function with the same duplicate params (ttl, redis, etc) which is also easier to test

Reply:
changed this quite a bit so it should be more familiar now. also moved the type definitions to the bottom so readers see the logic first

Reply:
much better